lib/Dialect/FIRRTL/Transforms/GrandCentralSignalMappings.cpp

```diff
@@ -546,27 +546,69 @@ FailureOr<bool> GrandCentralSignalMappingsPass::emitUpdatedMappings(
 
   // (2) Find input ports that are sinks, insert wires at instantiation points
   auto *instanceGraph = &getAnalysis<InstanceGraph>();
-  DenseSet<unsigned> forcedInputPorts;
+  llvm::SmallVector<unsigned, 32> forcedInputPorts;
+  llvm::SmallVector<unsigned, 32> forcedOutputPorts;
   for (auto &[_, mod, mappings] : results) {
     // Walk mappings looking for module ports that are being driven,
     // and gather their indices for fixing up.
     forcedInputPorts.clear();
+    forcedOutputPorts.clear();
     for (auto &mapping : mappings) {
       if (mapping.dir == MappingDirection::DriveRemote /* Sink */) {
         if (auto blockArg = mapping.localValue.dyn_cast<BlockArgument>()) {
           auto portIdx = blockArg.getArgNumber();
+          // Port may be driven multiple times along different instance paths
           if (mod.getPortDirection(portIdx) == Direction::In) {
-            // Port may be driven multiple times along different instance paths
-            forcedInputPorts.insert(portIdx);
+            forcedInputPorts.push_back(portIdx);
+          } else {
+            forcedOutputPorts.push_back(portIdx);
           }
         }
       }
     }
+    llvm::sort(forcedInputPorts);
+    forcedInputPorts.erase(
+        std::unique(forcedInputPorts.begin(), forcedInputPorts.end()),
+        forcedInputPorts.end());
+    llvm::sort(forcedOutputPorts);
+    forcedOutputPorts.erase(
+        std::unique(forcedOutputPorts.begin(), forcedOutputPorts.end()),
+        forcedOutputPorts.end());
 
     // Find all instantiations of this module, and replace uses of each driven
     // port with a wire that's connected to a wire that is connected to the
     // port. This is done to cause an 'assign' to be created, disconnecting
     // the forced input port's net from its uses.
```

Comment:
Does this comment still make sense given that `breakNet` is going to be used for output ports which are inside the current module?

Reply:
Good call, fixed this and the many other mentions of forcing input ports :+1: .